### PR TITLE
Fix printing both index signatures without type annotations and multi-parameter index signatures

### PR DESCRIPTION
**Pull request by cola119**
fixes https://github.com/prettier/prettier/issues/7225
- [x] I’ve added tests to confirm my change works.
- [ ] (If changing the API or CLI) I’ve documented the changes I’ve made (in the `docs/` directory)
- [x] (If the change is user-facing) I’ve added my changes to `changelog_unreleased/*/pr-XXXX.md` file following `changelog_unreleased/TEMPLATE.md`.
- [x] I’ve read the [contributing guidelines](https://github.com/prettier/prettier/blob/master/CONTRIBUTING.md).

**✨[Try the playground for this PR](https://prettier.io/playground-redirect)✨**

### CODE REVIEW

**Comment by cola119** [comment]
I guess I can fix https://github.com/prettier/prettier/issues/7224 quickly, but should I commit and push to this branch to avoid conflicts? Or is it better to create a new PR for that?

File: changelog_unreleased/typescript/pr-7228.md [hunk 1]
```diff
@@ -0,0 +1,11 @@
+#### Typescript: Remove unnecessary colons for TSIndexSignature without type parameters ([#7228](https://github.com/prettier/prettier/pull/7228) by [@cola119](https://github.com/cola119))
```
Comment (thorn0) [inline changelog_unreleased/typescript/pr-7228.md]:
```suggestion
#### Fix printing index signatures without type annotations ([#7228](https://github.com/prettier/prettier/pull/7228) by [@cola119](https://github.com/cola119))
```

File: tests/typescript_index_signature/__snapshots__/jsfmt.spec.js.snap [hunk 2]
```diff
@@ -11,11 +11,14 @@ class LocalStorage implements Storage {
   [key: string]: any;
 }
 
+type A = { [key: string] };
```
Comment (thorn0) [inline tests/typescript_index_signature/__snapshots__/jsfmt.spec.js.snap]: As an index signature without a type annotation is not valid TS, please move the test for this case to a separate file in `tests/typescript_error_recovery`.

**Comment by thorn0** [comment]
@cola119 I'd say it's okay to fix #7224 in this PR.

**Comment by cola119** [comment]
@thorn0 thanks your reviews. I modified the pointed out items and fixed https://github.com/prettier/prettier/issues/7224. 
Sorry in my poor English but please give me a review :bow:

**Comment by cola119** [comment]
@sosukesuzuki san,
I fixed to support long expressions and trailing comma. Please let me know if there are any problems.

**Pull request merged**